sympy/solvers/simplex.py

```diff
@@ -0,0 +1,821 @@
+        piv_rows.append(k)
+        r = _choose_pivot_row(A, B, piv_rows, c, Y)
+
+        M = _pivot(M, r, c)
+        X[c], Y[r] = Y[r], X[c]
+
+    # Phase 2: from a feasible solution, pivot to optimal
+    while True:
+        B = M[:-1, -1]
+        A = M[:-1, :-1]
+        C = M[-1, :-1]
+
+        # Choose a pivot column, c
+        piv_cols = []
+        piv_cols = [_ for _ in range(n) if C[_] < 0]
+        if not piv_cols:
+            break
+        c = sorted(piv_cols, key=lambda _: X[_])[0] # Bland's rule
+
+        # Choose a pivot row, r
+        piv_rows = [_ for _ in range(m) if A[_, c] > 0]
+        if not piv_rows:
+            raise UnboundedLPError(filldedent("""
+                Objective function can assume
+                arbitrarily large values!"""))
+        r = _choose_pivot_row(A, B, piv_rows, c, Y)
+
+        M = _pivot(M, r, c)
+        X[c], Y[r] = Y[r], X[c]
+
+    argmax = [None]*n
+    argmin_dual = [None]*m
+
+    for i, (v, n) in enumerate(X):
+        if v == False:
+            argmax[n] = 0
+        else:
+            argmin_dual[n] = M[-1, i]
+
+    for i, (v, n) in enumerate(Y):
+        if v == True:
+            argmin_dual[n] = 0
+        else:
+            argmax[n] = M[i, -1]
+
+    return -M[-1, -1], argmax, argmin_dual
+
+
+def _rel_as_nonpos(constr):
+    """return a (np, d, aux) where np is a list of nonpositive
+    expressions that represent the given constraints (possibly rewritten
+    in terms of auxilliary variables) expressible with nonnegative
+    symbols, and d is a dictionary mapping a given symbols to an
+    expression with an auxilliary variable. In some cases a symbol
+    will be used as part of the change of variables, e.g. x: x - z1
+    instead of x: z1 - z2.
+
+    If any constraint is False/empty, return None. Any constraint
+    involving a relationship with oo will create unbounded symbols
+    for all symbols present in the expression, e.g. ``x + y < oo``
+    will allow ``x`` and ``y`` to take on any value. (This is the
+    only type relationship which can be expressed as strict.)
+
+    Examples
+    ========
+
+    >>> from sympy import oo
+    >>> from sympy.solvers.simplex import _rel_as_nonpos
+    >>> from sympy.abc import x, y
+    >>> _rel_as_nonpos([x >= y])
+    ([-x + y], {}, [])
+    >>> _rel_as_nonpos([x > -oo])
+    ([], {x: _z1 - x}, [_z1])
+    >>> _rel_as_nonpos([x >= 3, x <= 5])
+    ([_z1 - 2], {x: _z1 + 3}, [_z1])
+    >>> _rel_as_nonpos([x <= 5])
+    ([], {x: 5 - _z1}, [_z1])
+    >>> _rel_as_nonpos([x >= 1])
+    ([], {x: _z1 + 1}, [_z1])
+    """
+    r = {}  # replacements to handle change of variables
+    np = []  # nonpositive expressions
+    aux = []  # auxilliary symbols added
+    ui = numbered_symbols('z', start=1, cls=Dummy) # auxilliary symbols
+    univariate = {}  # {x: interval} for univariate constraints
+    unbound = set()  # symbols designated as unbound
+
+    for i in constr:
+        if isinstance(i, AppliedBinaryRelation):
```

Comment:
I don't think it's necessary to handle predicate from the new assumptions anymore. We can get rid of this.